src/with-component.js

```diff
@@ -2,9 +2,10 @@ import { render } from 'preact';
 import { HTMLElement } from './util';
 import { withProps } from './with-props';
 import { withRender } from './with-render';
+import { withUnique } from './with-unique';
 
 export function withComponent (Base = HTMLElement) {
-  return class extends withRender(withProps(Base)) {
+  return class extends withUnique(withRender(withProps(Base))) {
```

Comment:
Did a simple perf test https://jsperf.com/prototype-chains. This basically means that each extension slows down by about 100k ops / sec, in Chrome, on a mid-2015 MacBook Pro.